### PR TITLE
Slightly better ndpi_strrstr implementation

### PR DESCRIPTION
**Pull request by 0xA50C1A1**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guidelines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Describe changes:

Nothing fancy, just added a couple of checks and made the code more readable.

@pasabanov @IvanNardi what do you think?

### CODE REVIEW

**Review by IvanNardi — APPROVED** [review]

File: src/lib/ndpi_utils.c [hunk 1]
```diff
@@ -3465,20 +3465,28 @@ int64_t ndpi_strtonum(const char *numstr, int64_t minval,
...
-  while(true) {
-    char *s = strstr(haystack, needle);
-    
-    if(s == NULL || s[0] == '\0')
+  if (!haystack || !needle) {
+    return NULL;
+  }
+
+  if (*needle == '\0') {
+    return (char*) haystack;
```
Comment (pasabanov) [inline src/lib/ndpi_utils.c]: Should return `(char*) haystack + strlen(haystack)` here.

**Review by pasabanov — CHANGES_REQUESTED** [review]

File: src/lib/ndpi_utils.c (hunk 1 above)
Comment (0xA50C1A1) [reply src/lib/ndpi_utils.c]: Done.

File: src/lib/ndpi_utils.c [hunk 2]
```diff
@@ -3465,20 +3465,28 @@ int64_t ndpi_strtonum(const char *numstr, int64_t minval,
...
-      ret = s;
-      haystack = &s[1]; /* Skip the first char */
     }
+
+    last_occurrence = (char*) current_pos;
+    haystack = current_pos + 1;
   }
 
-  return(ret);
+  return last_occurrence;
```
Comment (pasabanov) [inline src/lib/ndpi_utils.c]: I guess

```c
const char *last_occurrence = NULL;
...
last_occurrence = current_pos;
...
return (char*) last_occurrence;
```

might be slightly better.

**Review by pasabanov — CHANGES_REQUESTED** [review]

**Review by pasabanov — APPROVED** [review]

**Review by IvanNardi — APPROVED** [review]

**Comment by IvanNardi** [comment]
Thanks everyone

**Pull request merged**